GLSLang plugin does not compile with glslang v14

Current git GLSLang plugin does not compile with glslang v14 because this version made `class TType` opaque (its definition is not in the public headers anymore):

```
[  584s] + /usr/bin/c++ -DPlugin_GLSLangProgramManager_EXPORTS -I/home/abuild/rpmbuild/BUILD/ogre-14.1.2/PlugIns/GLSLang/include -I/home/abuild/rpmbuild/BUILD/ogre-14.1.2/OgreMain/include -I/home/abuild/rpmbuild/BUILD/ogre-14.1.2/build/include -Wall -Winit-self -Wcast-qual -Wwrite-strings -Wextra -Wundef -Wmissing-declarations -Wno-unused-parameter -Wshadow -Wno-missing-field-initializers -Wno-long-long -Wsuggest-override -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -msse -O3 -DNDEBUG -std=c++17 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -MD -MT PlugIns/GLSLang/CMakeFiles/Plugin_GLSLangProgramManager.dir/src/OgreGLSLang.cpp.o -MF CMakeFiles/Plugin_GLSLangProgramManager.dir/src/OgreGLSLang.cpp.o.d -o CMakeFiles/Plugin_GLSLangProgramManager.dir/src/OgreGLSLang.cpp.o -c /home/abuild/rpmbuild/BUILD/ogre-14.1.2/PlugIns/GLSLang/src/OgreGLSLang.cpp
[  586s] /home/abuild/rpmbuild/BUILD/ogre-14.1.2/PlugIns/GLSLang/src/OgreGLSLang.cpp: In member function 'virtual void Ogre::GLSLangProgram::prepareImpl()':
[  586s] /home/abuild/rpmbuild/BUILD/ogre-14.1.2/PlugIns/GLSLang/src/OgreGLSLang.cpp:379:17: error: invalid use of incomplete type 'const class glslang::TType'
[  586s]   379 |         if(utype->isOpaque())
[  586s]       |                 ^~
[  586s] In file included from /home/abuild/rpmbuild/BUILD/ogre-14.1.2/PlugIns/GLSLang/src/OgreGLSLang.cpp:11:
[  586s] /usr/include/glslang/Public/ShaderLang.h:146:7: note: forward declaration of 'class glslang::TType'
[  586s]   146 | class TType;
[  586s]       |       ^~~~~

```

--- a/PlugIns/GLSLang/src/OgreGLSLang.cpp
+++ b/PlugIns/GLSLang/src/OgreGLSLang.cpp
@@ -110,6 +110,70 @@ TBuiltInResource DefaultTBuiltInResource = {
     /* .maxDualSourceDrawBuffersEXT =  1,*/
     /* .limits = memset below*/
     };
+
+    /// punch through the private glslang API to get layoutLocation
+    struct TQualifier
+    {
+        // DANGER!! this must be kept in sync with the layout of glslang::TQualifier
+        const char*         semanticName;
+        int  storage      : 6;
+        int  builtIn      : 9;
+        int  declaredBuiltIn : 9;
+        int  precision    : 3;
+        bool invariant    : 1;
+        bool centroid     : 1;
+        bool smooth       : 1;
+        bool flat         : 1;
+        bool specConstant : 1;
+        bool nonUniform   : 1;
+        bool explicitOffset   : 1;
+        bool defaultBlock : 1;
+        bool noContraction: 1;
+        bool nopersp      : 1;
+        bool explicitInterp : 1;
+        bool pervertexNV  : 1;
+        bool perPrimitiveNV : 1;
+        bool perViewNV : 1;
+        bool perTaskNV : 1;
+        bool patch        : 1;
+        bool sample       : 1;
+        bool restrict     : 1;
+        bool readonly     : 1;
+        bool writeonly    : 1;
+        bool coherent     : 1;
+        bool volatil      : 1;
+        bool devicecoherent : 1;
+        bool queuefamilycoherent : 1;
+        bool workgroupcoherent : 1;
+        bool subgroupcoherent  : 1;
+        bool shadercallcoherent : 1;
+        bool nonprivate   : 1;
+        bool nullInit : 1;
+        bool spirvByReference : 1;
+        bool spirvLiteral : 1;
+
+        int layoutMatrix  : 3;
+        int layoutPacking : 4;
+        int layoutOffset;
+        int layoutAlign;
+
+        unsigned int layoutLocation             : 12;
+    };
+    struct TType
+    {
+        virtual ~TType() {}
+
+        // DANGER!! this must be kept in sync with the layout of glslang::TType
+        int basicType        : 8;
+        int vectorSize       : 4;
+        int matrixCols       : 4;
+        int matrixRows       : 4;
+        bool vector1         : 1;
+        bool coopmat         : 1;
+        TQualifier qualifier;
+
+        const TQualifier& getQualifier() const { return qualifier; }
+    };
 }
 
 namespace Ogre
@@ -375,8 +439,8 @@ void GLSLangProgram::prepareImpl()
     int blockIdx = -1;
     for(int i = 0; i < nuniforms; i++)
     {
-        auto utype = program.getUniformTType(i);
-        if(utype->isOpaque())
+        auto constType = mapToGCT(program.getUniformType(i));
+        if(GpuConstantDefinition::isSampler(constType) || constType == GCT_UNKNOWN)
             continue;
 
         blockIdx = program.getUniformBlockIndex(i);
@@ -393,11 +457,12 @@ void GLSLangProgram::prepareImpl()
             }
         }
 
+        auto utype = (const TType*)program.getUniformTType(i);
         GpuConstantDefinition def;
         def.logicalIndex = isUBO ? uoffset : utype->getQualifier().layoutLocation;
         def.arraySize = program.getUniformArraySize(i);
         def.physicalIndex = isUBO ? uoffset : mConstantDefs->bufferSize * 4;
-        def.constType = mapToGCT(program.getUniformType(i));
+        def.constType = constType;
         bool doPadding = isUBO && GpuConstantDefinition::getElementSize(def.constType, false) > 1;
         def.elementSize = GpuConstantDefinition::getElementSize(def.constType, doPadding);
 

